includes/Core/DI/DI_Services_Aware_Trait.php

```diff
@@ -0,0 +1,36 @@
+<?php
+/**
+ * Class Google\Site_Kit\Core\DI\DI_Services_Aware_Trait
+ *
+ * @package   Google\Site_Kit\Core\DI
+ * @copyright 2021 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://sitekit.withgoogle.com
+ */
+
+namespace Google\Site_Kit\Core\DI;
+
+/**
+ * Trait for DI services aware classes.
+ *
+ * @since n.e.x.t
+ */
+trait DI_Services_Aware_Trait {
+
+	/**
+	 * Gets DI service instance.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param string $name Service name.
+	 * @return mixed DI service on success, otherwise NULL.
+	 */
+	public function __get( $name ) {
```

Comment:
Not a fan of using magic getter here - it's unspecific and also a pain to integrate if the class itself wants to include its own. Let's make this a dedicated method, like `get_service`. I suppose you did this to make migration of the code easier, but I don't think we should go that route just because.

An acceptable path for me would be to keep this a magic getter just to simplify the initial PR and transition, and then have a follow-up issue to change the magic getter to a dedicated method, which would need to happen together with each class using e.g. `$this->get_service( 'assets' )` instead of `$this->assets`.

Reply:
Yes, let's keep it, for now, to make migration easier. I have moved the functionality into the new `get_di_entry( $name )` method and marked the `__get` method as deprecated.